query/src/main/java/org/infinispan/query/indexmanager/RemoteIndexingBackend.java

```diff
@@ -132,7 +134,8 @@ private void waitForReplacementBackend() {
    }
 
    private void sendCommand(ReplicableCommand command, List<LuceneWork> workList, boolean sync) {
-      rpcManager.invokeRemotely(recipients, command, rpcManager.getDefaultRpcOptions(sync));
+      CompletionStages.join(rpcManager.invokeCommand(recipients, command, VoidResponseCollector.ignoreLeavers(),
```

Comment:
you are ignoring the `sync` parameter.
The `recipients` can be replaced with a single `Address` (it doesn't change since it is assigned in the constructor)